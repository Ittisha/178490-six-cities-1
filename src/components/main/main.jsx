import * as React from 'react';
import PropTypes from 'prop-types';

import {CardList} from '../card-list/card-list';
import {CitiesMap} from '../cities-map/cities-map';
import {CityList} from '../city-list/city-list';
import {addPluralS} from '../../utils/addPluralS';
import ApartmentPropsShape from '../../props/apartment';

export const Main = ({apartments, city, cities, citiesCoords, handleCityChange, activeItem, setActiveItem, citiesZoom}) => {
  const apartmentsCoords = apartments.map(({id, coordinates, zoom}) => ({id, coordinates, zoom}));
  const apartmentsAmount = apartments.length;
  const placeWordForm = addPluralS(apartmentsAmount, `place`);
  return (
    <main className="page__main page__main--index">
      <h1 className="visually-hidden">Cities</h1>
      <CityList
        city={city.name}
        cities={cities}
        citiesCoords={citiesCoords}
        citiesZoom={citiesZoom}
        handleCityChange={handleCityChange}
      />
      <div className="cities__places-wrapper">
        <div className="cities__places-container container">
          <section className="cities__places places">
            <h2 className="visually-hidden">Places</h2>
            <b
              className="places__found">{`${apartmentsAmount} ${placeWordForm} to stay in ${city.name}`}</b>
            <form className="places__sorting" action="#" method="get">
              <span className="places__sorting-caption">Sort by</span>
              <span className="places__sorting-type" tabIndex="0">
                  Popular
                <svg className="places__sorting-arrow" width="7" height="4">
                  <use xlinkHref="#icon-arrow-select"></use>
                </svg>
              </span>
              <ul className="places__options places__options--custom places__options--opened">
                <li className="places__option places__option--active" tabIndex="0">Popular</li>
                <li className="places__option" tabIndex="0">Price: low to high
                </li>
                <li className="places__option" tabIndex="0">Price: high to low
                </li>
                <li className="places__option" tabIndex="0">Top rated first</li>
              </ul>
            </form>
            <div className="cities__places-list places__list tabs__content">
              <CardList apartments={apartments} setActiveOffer={setActiveItem}/>
            </div>
          </section>
          <div className="cities__right-section">
            <section className="cities__map map">
              <CitiesMap
                apartmentsCoords={apartmentsCoords}
                citiesZoom={citiesZoom}
                activeItem={activeItem}
                city={city}
              />
            </section>
          </div>
        </div>
      </div>
    </main>
  );
};

Main.propTypes = {
  apartments: PropTypes.arrayOf(ApartmentPropsShape).isRequired,
  city: PropTypes.shape({
    name: PropTypes.string.isRequired,
    coords: PropTypes.arrayOf(PropTypes.number).isRequired,
    zoom: PropTypes.number.isRequired,
  }).isRequired,
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  citiesCoords: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.number)).isRequired,
  citiesZoom: PropTypes.objectOf(PropTypes.number).isRequired,
  handleCityChange: PropTypes.func.isRequired,
  activeItem: ApartmentPropsShape,
  setActiveItem: PropTypes.func.isRequired,
};
