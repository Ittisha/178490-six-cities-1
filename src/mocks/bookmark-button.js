export const MOCK_PROPS = {
  offerId: 1,
  isInBookmarks: false,
  className: ``,
  width: 15,
  height: 12,
};
