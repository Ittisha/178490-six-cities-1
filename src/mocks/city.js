export const INIT_CITY = {
  name: `Amsterdam`,
  coords: [52.38333, 4.9],
  zoom: 12,
};

export const MOCK_CITIES = [`Amsterdam`, `Paris`];

export const MOCK_CITIES_COORDS = {
  Amsterdam: [52.38333, 4.9],
  Paris: [48.864716, 2.349014],
};

export const MOCK_CITIES_ZOOM = {
  Amsterdam: 17,
  Paris: 15,
};
