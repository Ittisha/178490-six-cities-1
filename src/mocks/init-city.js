export const INIT_CITY = {
  name: `Amsterdam`,
  coords: [52.38333, 4.9],
};
