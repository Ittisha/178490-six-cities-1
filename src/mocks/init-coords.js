export const INIT_COORDS = [52.38333, 4.9];
