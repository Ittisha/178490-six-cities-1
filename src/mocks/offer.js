export const OFFER = {
  "city": {
    "name": `Cologne`,
    "location": {
      "latitude": 50.938361,
      "longitude": 6.959974,
      "zoom": 13}},
  "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
  "images": [
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": false,
  "rating": 2.5,
  "type": `house`,
  "bedrooms": 2,
  "max_adults": 6,
  "price": 757,
  "goods": [
    `Coffee machine`,
    `Baby seat`,
    `Dishwasher`,
    `Fridge`,
    `Washing machine`,
    `Breakfast`,
  ],
  "host": {
    "id": 25,
    "name": `Angelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  "location": {
    "latitude": 50.945361,
    "longitude": 6.935974,
    "zoom": 16
  },
  "id": 1
};
