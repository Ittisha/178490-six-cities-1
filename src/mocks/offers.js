export const OFFERS = [
  {
    isPremium: true,
    isInBookmarks: false,
    title: `Beautiful & luxurious apartment at great location`,
    photoUrl: `img/apartment-01.jpg`,
    price: 120,
    type: `Apartment`,
    rating: 4.65,
    id: 1,
    coordinates: [52.3909553943508, 4.85309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
    bedrooms: 2,
    cityZoom: 15,
    description: `description`,
    goods: [`Something`],
    host: {
      id: 2,
      isPro: true,
      name: `Name`,
      avatar: `apartment-01.jpg`,
    },
    images: [`img/apartment-01.jpg`],
    maxAdults: 4,
    zoom: 16,
  },
  {
    isPremium: false,
    isInBookmarks: true,
    title: `Wood and stone place`,
    photoUrl: `img/room.jpg`,
    price: 80,
    type: `Private room`,
    rating: 4,
    id: 2,
    coordinates: [52.369553943508, 4.85309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
    bedrooms: 2,
    cityZoom: 15,
    description: `description`,
    goods: [`Something`],
    host: {
      id: 2,
      isPro: true,
      name: `Name`,
      avatar: `apartment-01.jpg`,
    },
    images: [`img/apartment-01.jpg`],
    maxAdults: 4,
    zoom: 16,
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Canal View Prinsengracht`,
    photoUrl: `img/apartment-02.jpg`,
    price: 132,
    type: `Apartment`,
    rating: 4,
    id: 3,
    coordinates: [52.3909553943508, 4.929309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
    bedrooms: 2,
    cityZoom: 15,
    description: `description`,
    goods: [`Something`],
    host: {
      id: 2,
      isPro: true,
      name: `Name`,
      avatar: `apartment-01.jpg`,
    },
    images: [`img/apartment-01.jpg`],
    maxAdults: 4,
    zoom: 16,
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Nice, cozy, warm big bed apartment`,
    photoUrl: `img/apartment-03.jpg`,
    price: 180,
    type: `Apartment`,
    rating: 4,
    id: 4,
    coordinates: [52.3809553943508, 4.939309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
    bedrooms: 2,
    cityZoom: 15,
    description: `description`,
    goods: [`Something`],
    host: {
      id: 2,
      isPro: true,
      name: `Name`,
      avatar: `apartment-01.jpg`,
    },
    images: [`img/apartment-01.jpg`],
    maxAdults: 4,
    zoom: 16,
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Nice, cozy, warm big bed apartment`,
    photoUrl: `img/apartment-03.jpg`,
    price: 100,
    type: `Apartment`,
    rating: 4.5,
    id: 5,
    coordinates: [48.866790, 2.378520],
    cityName: `Paris`,
    cityCoords: [48.864716, 2.349014],
    bedrooms: 2,
    cityZoom: 15,
    description: `description`,
    goods: [`Something`],
    host: {
      id: 2,
      isPro: true,
      name: `Name`,
      avatar: `apartment-01.jpg`,
    },
    images: [`img/apartment-01.jpg`],
    maxAdults: 4,
    zoom: 16,
  }

];

export const MOCK_FAVORITES = {
  'Amsterdam': [
    {
      isPremium: false,
      isInBookmarks: false,
      title: `Canal View Prinsengracht`,
      photoUrl: `img/apartment-02.jpg`,
      price: 132,
      type: `Apartment`,
      rating: 4,
      id: 3,
      coordinates: [52.3909553943508, 4.929309666406198],
      cityName: `Amsterdam`,
      cityCoords: [52.38333, 4.9],
      bedrooms: 2,
      cityZoom: 15,
      description: `description`,
      goods: [`Something`],
      host: {
        id: 2,
        isPro: true,
        name: `Name`,
        avatar: `apartment-01.jpg`,
      },
      images: [`img/apartment-01.jpg`],
      maxAdults: 4,
      zoom: 16,
    },
    {
      isPremium: false,
      isInBookmarks: false,
      title: `Nice, cozy, warm big bed apartment`,
      photoUrl: `img/apartment-03.jpg`,
      price: 180,
      type: `Apartment`,
      rating: 4,
      id: 4,
      coordinates: [52.3809553943508, 4.939309666406198],
      cityName: `Amsterdam`,
      cityCoords: [52.38333, 4.9],
      bedrooms: 2,
      cityZoom: 15,
      description: `description`,
      goods: [`Something`],
      host: {
        id: 2,
        isPro: true,
        name: `Name`,
        avatar: `apartment-01.jpg`,
      },
      images: [`img/apartment-01.jpg`],
      maxAdults: 4,
      zoom: 16,
    },
  ]
};
