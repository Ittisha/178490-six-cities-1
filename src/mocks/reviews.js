export const MOCK_REVIEWS = [
  {
    id: 1,
    user: {
      avatarUrl: ``,
      id: 1,
      isPro: false,
      name: ``,
    },
    rating: 4,
    comment: ``,
    date: ``,
  },
  {
    id: 2,
    user: {
      avatarUrl: ``,
      id: 5,
      isPro: false,
      name: ``,
    },
    rating: 3,
    comment: ``,
    date: ``,
  }
];

export const MOCK_REVIEW = {
  comment: ``,
  id: 1,
  rating: `3`,
};
