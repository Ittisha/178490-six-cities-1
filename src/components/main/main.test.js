import * as React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Main} from './main';
import {OFFERS} from '../../mocks/offers';
import {INIT_CITY} from '../../mocks/init-city';

configure({adapter: new Adapter()});

const cities = [`Amsterdam`, `Paris`];
const citiesCoords = {
  Amsterdam: [52.38333, 4.9],
  Paris: [48.864716, 2.349014],
};
const citiesZoom = {
  Amsterdam: 17,
  Paris: 15,
};

it(`Main component is rendered properly`, () => {
  const tree = shallow(
      <Main
        apartments={OFFERS}
        city={INIT_CITY}
        cities={cities}
        citiesCoords={citiesCoords}
        citiesZoom={citiesZoom}
        handleCityChange={() => {}}
        setActiveItem={() => {}}
      />);

  expect(tree).toMatchSnapshot();
});
