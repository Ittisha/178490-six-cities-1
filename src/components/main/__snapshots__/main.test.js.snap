// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component is rendered properly 1`] = `
<main
  className="page__main page__main--index false"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <CityList
    cities={
      Array [
        "Amsterdam",
        "Paris",
      ]
    }
    citiesCoords={
      Object {
        "Amsterdam": Array [
          52.38333,
          4.9,
        ],
        "Paris": Array [
          48.864716,
          2.349014,
        ],
      }
    }
    citiesZoom={
      Object {
        "Amsterdam": 17,
        "Paris": 15,
      }
    }
    city="Amsterdam"
    handleCityChange={[Function]}
  />
  <div
    className="cities__places-wrapper"
  >
    <div
      className="cities__places-container false container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          5 places to stay in Amsterdam
        </b>
        <WithActiveItem
          onOptionChange={[Function]}
        />
        <div
          className="cities__places-list places__list tabs__content"
        >
          <CardList
            apartments={
              Array [
                Object {
                  "bedrooms": 2,
                  "cityCoords": Array [
                    52.38333,
                    4.9,
                  ],
                  "cityName": "Amsterdam",
                  "cityZoom": 15,
                  "coordinates": Array [
                    52.3909553943508,
                    4.85309666406198,
                  ],
                  "description": "description",
                  "goods": Array [
                    "Something",
                  ],
                  "host": Object {
                    "avatar": "apartment-01.jpg",
                    "id": 2,
                    "isPro": true,
                    "name": "Name",
                  },
                  "id": 1,
                  "images": Array [
                    "img/apartment-01.jpg",
                  ],
                  "isInBookmarks": false,
                  "isPremium": true,
                  "maxAdults": 4,
                  "photoUrl": "img/apartment-01.jpg",
                  "price": 120,
                  "rating": 4.65,
                  "title": "Beautiful & luxurious apartment at great location",
                  "type": "Apartment",
                  "zoom": 16,
                },
                Object {
                  "bedrooms": 2,
                  "cityCoords": Array [
                    52.38333,
                    4.9,
                  ],
                  "cityName": "Amsterdam",
                  "cityZoom": 15,
                  "coordinates": Array [
                    52.369553943508,
                    4.85309666406198,
                  ],
                  "description": "description",
                  "goods": Array [
                    "Something",
                  ],
                  "host": Object {
                    "avatar": "apartment-01.jpg",
                    "id": 2,
                    "isPro": true,
                    "name": "Name",
                  },
                  "id": 2,
                  "images": Array [
                    "img/apartment-01.jpg",
                  ],
                  "isInBookmarks": true,
                  "isPremium": false,
                  "maxAdults": 4,
                  "photoUrl": "img/room.jpg",
                  "price": 80,
                  "rating": 4,
                  "title": "Wood and stone place",
                  "type": "Private room",
                  "zoom": 16,
                },
                Object {
                  "bedrooms": 2,
                  "cityCoords": Array [
                    52.38333,
                    4.9,
                  ],
                  "cityName": "Amsterdam",
                  "cityZoom": 15,
                  "coordinates": Array [
                    52.3909553943508,
                    4.929309666406198,
                  ],
                  "description": "description",
                  "goods": Array [
                    "Something",
                  ],
                  "host": Object {
                    "avatar": "apartment-01.jpg",
                    "id": 2,
                    "isPro": true,
                    "name": "Name",
                  },
                  "id": 3,
                  "images": Array [
                    "img/apartment-01.jpg",
                  ],
                  "isInBookmarks": false,
                  "isPremium": false,
                  "maxAdults": 4,
                  "photoUrl": "img/apartment-02.jpg",
                  "price": 132,
                  "rating": 4,
                  "title": "Canal View Prinsengracht",
                  "type": "Apartment",
                  "zoom": 16,
                },
                Object {
                  "bedrooms": 2,
                  "cityCoords": Array [
                    52.38333,
                    4.9,
                  ],
                  "cityName": "Amsterdam",
                  "cityZoom": 15,
                  "coordinates": Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                  "description": "description",
                  "goods": Array [
                    "Something",
                  ],
                  "host": Object {
                    "avatar": "apartment-01.jpg",
                    "id": 2,
                    "isPro": true,
                    "name": "Name",
                  },
                  "id": 4,
                  "images": Array [
                    "img/apartment-01.jpg",
                  ],
                  "isInBookmarks": false,
                  "isPremium": false,
                  "maxAdults": 4,
                  "photoUrl": "img/apartment-03.jpg",
                  "price": 180,
                  "rating": 4,
                  "title": "Nice, cozy, warm big bed apartment",
                  "type": "Apartment",
                  "zoom": 16,
                },
                Object {
                  "bedrooms": 2,
                  "cityCoords": Array [
                    48.864716,
                    2.349014,
                  ],
                  "cityName": "Paris",
                  "cityZoom": 15,
                  "coordinates": Array [
                    48.86679,
                    2.37852,
                  ],
                  "description": "description",
                  "goods": Array [
                    "Something",
                  ],
                  "host": Object {
                    "avatar": "apartment-01.jpg",
                    "id": 2,
                    "isPro": true,
                    "name": "Name",
                  },
                  "id": 5,
                  "images": Array [
                    "img/apartment-01.jpg",
                  ],
                  "isInBookmarks": false,
                  "isPremium": false,
                  "maxAdults": 4,
                  "photoUrl": "img/apartment-03.jpg",
                  "price": 100,
                  "rating": 4.5,
                  "title": "Nice, cozy, warm big bed apartment",
                  "type": "Apartment",
                  "zoom": 16,
                },
              ]
            }
            cardClass="cities__place-card"
            cardListClass="cities__places-list tabs__content"
            imageWrapperClass="cities__image-wrapper"
            setActiveOffer={[Function]}
          />
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <CitiesMap
            apartmentsCoords={
              Array [
                Object {
                  "coordinates": Array [
                    52.3909553943508,
                    4.85309666406198,
                  ],
                  "id": 1,
                  "zoom": 16,
                },
                Object {
                  "coordinates": Array [
                    52.369553943508,
                    4.85309666406198,
                  ],
                  "id": 2,
                  "zoom": 16,
                },
                Object {
                  "coordinates": Array [
                    52.3909553943508,
                    4.929309666406198,
                  ],
                  "id": 3,
                  "zoom": 16,
                },
                Object {
                  "coordinates": Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                  "id": 4,
                  "zoom": 16,
                },
                Object {
                  "coordinates": Array [
                    48.86679,
                    2.37852,
                  ],
                  "id": 5,
                  "zoom": 16,
                },
              ]
            }
            citiesZoom={
              Object {
                "Amsterdam": 17,
                "Paris": 15,
              }
            }
            city={
              Object {
                "coords": Array [
                  52.38333,
                  4.9,
                ],
                "name": "Amsterdam",
                "zoom": 12,
              }
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;
